invisible_cities/io/mcinfo_io_test.py

```diff
@@ -206,54 +89,187 @@ def test_copy_mc_info_which_events_out_of_range(ICDATADIR, config_tmpdir):
+            assert new_tbls[key].notna().all().sum() == new_tbls[key].shape[1]
+            assert_dataframes_equal(new_tbls[key].loc[:, old_notnan],
+                                    old_tbls[key].loc[:, old_notnan],
+                                    check_types=False)
+
+
+def test_read_mc_tables(mc_particle_and_hits_nexus_data_new):
+    file_in, *_ = mc_particle_and_hits_nexus_data_new
+
+    input_tbls  = get_mc_tbl_list          (file_in)
+    all_evt     = get_event_numbers_in_file(file_in)
+
+    shape_dict  = {}
+    with tb.open_file(file_in) as h5in:
+        for tbl in h5in.root.MC:
+            shape_dict[MCTableType[tbl.name]] = (tbl.shape[0], len(tbl.cols))
+
+    tbl_dict    = read_mc_tables(file_in, all_evt)
+    for tbl, shp in shape_dict.items():
+        assert tbl in tbl_dict.keys()
+        assert shp == tbl_dict[tbl].shape
+
+
+def test_mc_writer(mc_particle_and_hits_nexus_data_new, config_tmpdir):
+    file_in, *_ = mc_particle_and_hits_nexus_data_new
+    file_out    = os.path.join(config_tmpdir, 'dummy_out.h5')
+
+    input_tbls  = get_mc_tbl_list          (file_in)
+    all_evt     = get_event_numbers_in_file(file_in)
+
+    tbl_dict    = read_mc_tables(file_in, all_evt)
+    with tb.open_file(file_out, 'w') as h5out:
+        mcwrite = mc_writer(h5out)
+        mcwrite(tbl_dict)
 
-    assert len(mchits_dict[evt_number]) == number_of_hits
+    saved_tbls = get_mc_tbl_list(file_out)
+    assert np.all(input_tbls == saved_tbls)
 
 
-def test_load_mchits(mc_particle_and_hits_nexus_data):
-    efile, _, _, _, _, _, _, X, Y, Z, E, t = mc_particle_and_hits_nexus_data
+def test_mc_writer_oldformat(mc_sensors_nexus_data, config_tmpdir):
+    file_in, *_ = mc_sensors_nexus_data
+    file_out    = os.path.join(config_tmpdir, 'dummy_out.h5')
 
-    mchits_dict = load_mchits(efile)
-    hX = [hit.X    for hit in mchits_dict[0]]
-    hY = [hit.Y    for hit in mchits_dict[0]]
-    hZ = [hit.Z    for hit in mchits_dict[0]]
-    hE = [hit.E    for hit in mchits_dict[0]]
-    ht = [hit.time for hit in mchits_dict[0]]
+    input_tbls  = get_mc_tbl_list          (file_in)
+    all_evt     = get_event_numbers_in_file(file_in)
 
-    assert np.allclose(X, hX)
-    assert np.allclose(Y, hY)
-    assert np.allclose(Z, hZ)
-    assert np.allclose(E, hE)
-    assert np.allclose(t, ht)
+    tbl_dict    = read_mc_tables(file_in, all_evt, 'new', 6400)
+    with tb.open_file(file_out, 'w') as h5out:
+        mcwrite = mc_writer(h5out)
+        mcwrite(tbl_dict)
+
+    saved_tbls = get_mc_tbl_list(file_out)
+    ## The extents and events tables are not saved.
+    assert len(input_tbls) == len(saved_tbls) + 2
+    assert MCTableType.configuration in saved_tbls
+    assert MCTableType.hits          in saved_tbls
+    assert MCTableType.particles     in saved_tbls
+    assert MCTableType.sns_positions in saved_tbls
+    assert MCTableType.sns_response  in saved_tbls
+
+
+@fixture(scope = 'module')
```

Comment:
if  `mc_particle_and_hits_nexus_data` is only used in this module you can move it here from conftest.py